### ISSUE TITLE
Enabling group-updates causes exabgp to group multiple NLRI with multiple MP_(UN)REACH in one UPDATE message

### ISSUE THREAD

**Issue by sonixyznitesh** [Exa-Networks/exabgp#742]
##### ISSUE TYPE
 - Bug Report

##### OS
```
Linux
CentOS 6.8
```

##### VERSION
```
ExaBGP : 3.4.20
Python : 2.6.6 (r266:84292, Oct 31 2017, 08:19:29)  [GCC 4.4.7 20120313 (Red Hat 4.4.7-3)]
Uname  : #1 SMP Tue Oct 31 08:18:37 PDT 2017
```

##### ENVIRONMENT 
```
empty
```

##### CONFIGURATION
```
group template {
	group-updates;
	neighbor ::ffff:10.128.10.128 {
		local-address ::ffff:10.128.10.169;
		local-as 60000;
		peer-as 60000;
		family {
			ipv4 unicast;
			ipv4 flow;
			ipv6 unicast;
			ipv6 flow;
		}
		router-id 1.1.1.1;
		hold-time 90;
		capability {
			asn4;
			route-refresh;
		}
	}
}

```

##### SUMMARY

ExaBGP groups multiple NLRIs (flow spec routes) in a single UPDATE message with 'group-updates' attribute enabled.

However, if the multiple flow spec routes have different next hops, this causes multiple MP_REACH (or MP_UNREACH) path attributes in the UPDATE message which gets rejected by the upstream router with following error message:

"UPDATE from A.B.C.D contains duplicate MP_REACH attribute"

Per RFC https://tools.ietf.org/html/draft-ietf-idr-error-handling-19, section 5.1

```
An UPDATE message MUST NOT contain more than one of the following:
      non-empty Withdrawn Routes field, non-empty Network Layer
      Reachability Information field, MP_REACH_NLRI attribute, and
      MP_UNREACH_NLRI attribute.
```

##### STEPS TO REPRODUCE

With above configuration, send following routes:

```
exabgp: 868    processes     Command from process announce-routes : announce flow route { match { destination 11.11.12.0/24; } then { redirect 11.11.11.199; } } 
exabgp: 868    processes     Command from process announce-routes : announce flow route { match { destination 11.11.13.0/24; } then { redirect 11.11.11.198; } } 
```

##### EXPECTED RESULTS

Multiple flow spec NLRI with multiple MP_REACH attributes are not encoded in single BGP UPDATE message. Only multiple NLRI (flow spec) are encoded in single BGP UPDATE as long as they have same next hop (MP_REACH or MP_UNREACH) path attribute.

Note that I am not sure whether multiple NLRI itself are also allowed even if they have same MP_REACH in same UPDATE message.

##### ACTUAL RESULTS
Multiple flow spec NLRI with multiple MP_REACH attributes are encoded in single BGP UPDATE message (with group-updates enabled). This violates the RFC standard and breaks upstream router's handling of BGP UPDATE

**Comment by thomas-mangin** [comment]
This issue was already fixed in 3.4.21 (issue #493)

**Comment by thomas-mangin** [comment]
Thank you for the good report and sorry for not spotting earlier. Released 3.4.22 today - enjoy !

**Comment by sonixyznitesh** [comment]
Hello Thomas,

Thank you, first of all, for taking a quick look and providing a timely resolution!

Just to confirm, the fix (for #493) should work for IPv4 family AND FlowSepc NLRIs too - correct?
Anyways, I will upgrade to 3.4.22 soon and let you know how it goes!

And lastly, thank you for authoring ExaBGP - we have been using it for injecting FlowSpec routes and it has been almost flawless to-date :)

Regards
/Nitesh

**Comment by thomas-mangin** [comment]
Sent from my iPad

> On 4 Nov 2017, at 18:44, sonixyznitesh <notifications@github.com> wrote:
> 
> Hello Thomas,
> 
> Thank you, first of all, for taking a quick look and providing a timely resolution!
> 
> Just to confirm, the fix (for #493) should work for IPv4 family AND FlowSepc NLRIs too - correct?
> 
Correct it only groups IPv4 unicast routes.
> Anyways, I will upgrade to 3.4.22 soon and let you know how it goes!
> 
> And lastly, thank you for authoring ExaBGP - we have been using it for injecting FlowSpec routes and it has been almost flawless to-date :)
> 
Nice to hear.

**Comment by sonixyznitesh** [comment]
Upgraded to 3.4.22 today and tested out the fix. Works as expected!

Thanks!
/Nitesh
